server/app/src/androidTest/java/sh/calaba/instrumentationbackend/entrypoint/AndroidInstrumentationStartup.java

```diff
@@ -180,4 +188,19 @@ private String detectMainActivity(StatusReporter statusReporter, String targetPa
             throw new RuntimeException(e);
         }
     }
+
+    private void reportFailure(String message) {
+        hasReportedFailure = true;
```

Comment:
Why do we need this?

Reply:
It's a duplicate variable from old StatusReporter. Taking into account that we are not able to use it for sdk > 30 (android 11), I moved this variable here for compatibility.
I'm not sure that it is needed at all but I probably just don't know possible drawbacks of removing it so I decided to keep it as-is.